bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/di/PresentationModule.kt

```diff
@@ -1,10 +1,38 @@
 package network.bisq.mobile.presentation.di
 
+import androidx.compose.runtime.mutableStateOf
+import androidx.navigation.NavController
+import androidx.navigation.NavHostController
 import network.bisq.mobile.presentation.MainPresenter
 import network.bisq.mobile.presentation.ui.AppPresenter
+import network.bisq.mobile.presentation.ui.uicases.startup.IOnboardingPresenter
+import network.bisq.mobile.presentation.ui.uicases.startup.ISplashPresenter
+import network.bisq.mobile.presentation.ui.uicases.startup.OnBoardingPresenter
+import network.bisq.mobile.presentation.ui.uicases.startup.SplashPresenter
+import network.bisq.mobile.presentation.ui.uicases.GettingStartedPresenter
+import network.bisq.mobile.presentation.ui.uicases.IGettingStarted
+import org.koin.core.qualifier.named
 import org.koin.dsl.bind
 import org.koin.dsl.module
 
 val presentationModule = module {
+
+    single<NavHostController> { error("NavController has not been initialized yet") }
```

Comment:
Quick question on this one, why the approach of erroring to the NavHostController request instead of using the single {} scope to configure it and return it?

Not opposing just want to understand what's the idea behind this - thank you :pray: 

- note I had to fix crashes of UI components asking for NavHostController using named() koin function - please review @nostrbuddha

Reply:
Yeah. Missed using `named()` in Tab screens and wasted time, looking for the reason for runtime errors elsewhere. Thanks for finding and fixing it.

For your question,
It's actually not needed now. I will remove it.

I had an error back then, when I tried to access NavHostController, while assigning the the value to Koin DI inside LaunchedEffect. To fix that, added that `isNavControllerSet` state variable and render the components, once after this var is set.
At that time, just added this as a Guard logic. But it didn't work that way either!

Also `NavHostController` can be initialized only inside a `@Composable`, hence the current logic of doing that in `App.kt` and assigning that to Koin DI inside `LaunchedEffect`. Can't initialize it directly in `PresentationModule.kt`, since it's not a `@Composable`.